### ISSUE TITLE
Create Particle class method to find number of valence electrons

### ISSUE THREAD

**Issue by namurphy** [PlasmaPy/PlasmaPy#283]
During the code review session yesterday, @lemmatum suggested that we include an attribute in the `Particle` class that would return the number of [valence electrons](https://en.wikipedia.org/wiki/Valence_electron) for a neutral atom or ion.  This attribute should be useful for ionized species as well (e.g., a Fe<sup>16+</sup> ion will have a closed shell).  I'm not a chemist, so it would be really helpful to get feedback from people who study plasma chemistry and/or atomic physicists in order to figure out the best way to structure this and what conventions we should use, especially for highly ionized species.

**Comment by ritiek** [comment]
> it would be really helpful to get feedback from people who study plasma chemistry and/or atomic physicists in order to figure out the best way to structure this and what conventions we should use, especially for highly ionized species.

Yep, also how should we go about Transition metals, Lanthanoids and Actinoids as they have kinda unpredictable valence electrons? Should we just make a make a special case for each of them?

And what happens to their ions? How will the number of valence electrons change?

For example, Rh⁺ ion (atomic no. 45, neutral electron configuration: 2, 8, 18, 16, 1)? Will it follow same configuration as neutral Ru atom (atomic. no 44, neutral electron configuration: 2, 8, 18, 15, 1) or a different configuration of 2, 8, 18, 16? The number of valence electrons differ in both these cases.
